### DISCUSSION TITLE
Faster build time and smaller cache size

### DISCUSSION THREAD

**Discussion by MNITD** [parcel-bundler/parcel#4643] [General] [answered]
Hello!

I am considering parcel 2 as a substitution of current custom webpack configuration on react+typescript project that I work on because I like zero-configuration idea and react-refresh support 

But I found several points that are critical for the final decision about migration to parcel:

- core parcel does not support es features like optional chaining, class props, etc ( can be fixed just by installing relative babel-plugins)
- .parcel-cache folder growth up to 800 MB and might grow further 
- source-map enabled leads to out of heap memory error after several hot reloads
- build time takes from 2 up to 10 sec even if using cache (compering to 800 ms on my custom webpack config)

Could you please consult me about parcel and what can I do to have both source-map and hot-reload as well as an increase in speed and smaller cache?

P.S. I have no idea what I can do with cache growth, but I think I might do something with speed by using ts-loader (https://github.com/TypeStrong/ts-loader) instead of babel and node-sass instead of dart-sass, as this stack is using in my custom config.

**Comment by DeMoorJasper** [comment, marked as answer]
@MNITD the sourcemaps out of heap sounds like a bug could you report that with a reproducible or something?

Fixing the cache folder size is already on our roadmap as far as I know

A faster build time than 800ms is probably very unlikely except on hot-builds which shouldn't take any longer than a couple milliseconds. Most of our buildtime is spent by Babel and is kind of out of our control, we're looking into optimising this by using alternatives like swc or esbuild

node-sass is unreliable in multithreaded/multiprocess applications so we don't use it. I think we still allow it if it's installed although I'm not sure

**Reply by MNITD** [reply]
> the sourcemaps out of heap sounds like a bug could you report that with a reproducible or something

Sure, I will try to create a bug report on the next week
